frontend/src/components/BalancePresentation.js

```diff
@@ -5,13 +5,18 @@ import "../styles/BalancePresentation.css";
 class BalancePresentation extends React.Component {
   constructor(props) {
     super(props);
+
     this.state = {
-      salary: 5000,
+      salary: this.props.salary,
       saving: 1000,
       expenses: 2800,
       balance: 1200,
     };
   }
+  componentWillReceiveProps(nextProps) {
+    this.setState({ salary: nextProps.salary });
+  }
+
```

Comment:
This seems unnecessarily complex, let's rather just use this.props.salary directly in the rendering.
```js
<tr>
  <td>Total salary</td>
  <td>{this.props.salary} €</td>
</tr>
```

Reply:
changes done.

Reply:
9a3fedac0085772f77a7003f22ef18776a761afa OK